pkg/authentication/handlers.go

```diff
@@ -123,6 +169,37 @@ func (a *API) handleLogin(w http.ResponseWriter, r *http.Request) {
 	}
 
 	_ = json.NewEncoder(w).Encode(tokens)
+
+}
+
+func (a *API) nonceCheck(r *http.Request, idToken *Principal) error {
+	nonce := a.cookieManager.GetNonceCookie(r)
+	if nonce == "" {
+		a.logger.Error(nonceCookieNotFoundMsg)
+		return nonceCookieNotFoundErr
+	}
+
+	if idToken.Nonce != nonce {
+		a.logger.Error(idTokenNonceMatchMsg)
+		return idTokenNonceMatchErr
+	}
+
+	return nil
+}
+
+func (a *API) stateCheck(r *http.Request, state string) error {
```

Comment:
nitpick: I prefer `checkState`, that way is it is clearer that this function checks something